framework/wazuh/core/stats.py

```diff
@@ -2,23 +2,215 @@
+
+    Returns
+    -------
+    array
+        Hours and interactions for each week day.
+    """
+    weekly_results = []
+    for i in range(7):
+        hours = []
+        interactions = 0
+        for j in range(25):
+            try:
+                wfile = open(f'{common.stats_path}/weekly-average/{str(i)}/{str(j)}')
+                data = wfile.read()
+                if j == 24:
+                    interactions = int(data)
+                else:
+                    hours.append(int(data))
+                wfile.close()
+            except IOError:
+                if i < 24:
+                    hours.append(0)
+        weekly_results.append({DAYS[i]: {'hours': hours, 'interactions': interactions}})
+
+    return weekly_results
+
+
+def totals_(date):
+    """Statistical information for the current or specified date.
+
+    Parameters
+    ----------
+    date: date
+        Date object with the date value of the stats.
+
+    Returns
+    -------
+    tuple
+        Failed: boolean that represents if data parsing failed. Its value is always `False`.
+        Affected: array of dictionaries. Each dictionary represents an hour.
+
+    Raises
+    ------
+    WazuhError
+        Raised on `IOError`.
+    """
+    try:
+        stat_filename = os.path.join(
+            common.stats_path, "totals", str(date.year), MONTHS[date.month - 1],
+            f"ossec-totals-{date.strftime('%d')}.log")
+        stats = open(stat_filename, 'r')
+    except IOError:
+        raise WazuhError(1308, extra_message=stat_filename)
+    alerts = []
+    affected = []
+    for line in stats:
+        data = line.split('-')
+        if len(data) == 4:
+            sigid = int(data[1])
+            level = int(data[2])
+            times = int(data[3])
+            alert = {'sigid': sigid, 'level': level, 'times': times}
+            alerts.append(alert)
+        else:
+            data = line.split('--')
+            if len(data) != 5:
+                if len(data) in (0, 1):
+                    continue
+                else:
+                    return True, affected
+            hour = int(data[0])
+            total_alerts = int(data[1])
+            events = int(data[2])
+            syscheck = int(data[3])
+            firewall = int(data[4])
+            affected.append(
+                {'hour': hour,
+                 'alerts': alerts,
+                 'totalAlerts': total_alerts,
+                 'events': events,
+                 'syscheck': syscheck,
+                 'firewall': firewall
+                 })
+            alerts = []
+
+    return False, affected
+
+
+def get_daemons_stats_(filename):
+    """Get daemons stats from an input file.
+
+    Parameters
+    ----------
+    filename: str
+        Full path of the file to get information.
+
+    Returns
+    -------
+    array
+        Stats of the input file.
+
+    Raises
+    ------
+    WazuhError
+        Raised if file does not exist.
+    """
+    try:
```

Comment:
All this piece of code needs to be evaluated, it seems very weird, what is its exact purpose? can it be done any other way?

Reply:
I'm currently working on refactoring this block in a simpler way.